src/wp-admin/includes/class-wp-plugin-dependencies.php

```diff
@@ -0,0 +1,634 @@
+					esc_html( $parents_names[0] )
+				);
+
+			$this->inline_plugin_row_notice( $notice_contents, 'info', $plugin_file );
+		}
+
+		// Add extra info to parents.
+		if ( $has_dependencies ) {
+			$style = is_rtl() ? 'border-top:none;border-left:none' : 'border-top:none;border-right:none';
+			if ( $pending_activation ) {
+				if ( $in_circular_dependency ) {
+					$this->inline_plugin_row_notice(
+						sprintf(
+							/* translators: %s: plugin name. */
+							esc_html__( 'Warning: Circular dependencies detected. Plugin "%s" has unmet dependencies.' ),
+							esc_html( $plugin_data['Name'] )
+						),
+						'warning',
+						$plugin_file
+					);
+				} else {
+					$this->inline_plugin_row_notice(
+						sprintf(
+							/* translators: %s: plugin name. */
+							esc_html__( 'Plugin "%s" has unmet dependencies. Once all required plugins are installed the plugin will be automatically activated. Alternatively you can cancel the activation of this plugin by clicking on the "cancel activation request" link above.' ),
+							esc_html( $plugin_data['Name'] )
+						),
+						'warning',
+						$plugin_file
+					);
+				}
+			} elseif ( ! $is_plugin_active ) {
+				$dependencies_human_readable = array();
+				foreach ( $dependencies as $dependency ) {
+					$plugin_file = $this->get_plugin_file_from_slug( $dependency['slug'] );
+					if ( $plugin_file ) {
+						$dependencies_human_readable[] = get_plugin_data( WP_PLUGIN_DIR . '/' . $plugin_file )['Name'];
+					} else {
+						$dependencies_human_readable[] = $dependency['slug'];
+					}
+				}
+				$this->inline_plugin_row_notice(
+					sprintf(
+						/* translators: %1$s: plugin name. %2$s: plugin requirements, comma-separated. */
+						esc_html__( 'Plugin "%1$s" depends on the following plugin(s): %2$s' ),
+						esc_html( $plugin_data['Name'] ),
+						esc_html( implode( ', ', $dependencies_human_readable ) )
+					),
+					'info',
+					$plugin_file
+				);
+			}
+		}
+	}
+
+	/**
+	 * Generate the contents of an inline plugin row notice.
+	 *
+	 * @since 5.9.0
+	 * @access protected
+	 */
+	protected function inline_plugin_row_notice( $contents = '', $notice_type = 'info', $plugin_file = '' ) {
+		$is_plugin_active = is_plugin_active( $plugin_file );
+		$tr_class         = $is_plugin_active ? 'plugin-dependencies-tr active' : 'plugin-dependencies-tr';
+		$td_class         = 'plugin-dependencies colspanchange';
+		$colspan          = (int) _get_list_table( 'WP_Plugins_List_Table', array( 'screen' => get_current_screen() ) )->get_column_count();
+		?>
+		<tr class="<?php echo esc_attr( $tr_class ); ?>">
+			<td class="<?php echo esc_attr( $td_class ); ?>" colspan="<?php echo esc_attr( $colspan ); ?>" style="padding:0">
+				<div class="dependencies-message notice inline notice-<?php echo esc_attr( $notice_type ); ?> notice-alt" style="margin:0;border-top:none;border-<?php echo is_rtl() ? 'left' : 'right'; ?>:none;">
+					<p><?php echo $contents; ?></p>
```

Comment:
Not 100% sure, but I think that this will throw a WPCS escaping error, so you can add a `// phpcs:ignore WordPress.Security.EscapeOutput This output is escaped beforehand.`, since in all the cases where you are passing content to this method, it's escaped.